core/db/db_setup.js

```diff
@@ -5,5 +5,7 @@ db.createCollection('orderbooks');
 db.createCollection('messages');
 
 db.orderbooks.createIndex({ instrument: 1, timestamp: 1 });
+
 db.messages.createIndex({ instrument: 1, timestamp: 1 });
 db.messages.createIndex({ instrument: 1, sod_offset: 1 });
+db.messages.createIndex({ instrument: 1, sod_offset: -1 });
```

Comment:
Why is this necessary?

Reply:
We have to sort by sod_offset in both ways for pagination. Direction of the index matters for composite keys in mongodb. https://stackoverflow.com/questions/10329104/why-does-direction-of-index-matter-in-mongodb